styhead/rz sbc: add renesas core images

tiennguyenuh opened the pull request:
Add recipes for Renesas core images including:
- renesas-core-image-cli
- renes-core-image-weston
- renesas-quickboot-cli
- renesas-quickboot-wayland

tiennguyenuh committed:
rz-sbc: modify renesas-core-image-weston and install kernel modules by default

Changes include:
    - Install kernel module in renesas-base image.
    - Modify renesas core-image-weston based on qt version.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: add renesas core images and quickboot images

Changes include:
    - Add Renesas quickboot wayland and cli images.
    - Retain networking stack by using script that allows user to enable
      or disable network-related services.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: add python libraries and apt source list

Changes include:
    - Add python libraries and pip.
    - Install default APT sources list using Ubuntu 24.04 repositories.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

preetam-reddy commented on recipes-extend/systemd-network-control/files/quickboot-wayland/enable_networking_stack.sh:
```diff
@@ -0,0 +1,204 @@
+# Function to remove symlinks for systemd udev services
+unmask_udev_services() {
+    echo "Removing systemd udev service symlinks..."
+    rm -f /etc/systemd/system/systemd-udevd
```
you should be able to use `systemctl disable <service>` and `systemctl mask <service>`.
It's not a good idea to delete systemd service url's manually.

tiennguyenuh replied:
This is a null symlink created by the `disable_networking_stack.sh` script, for example: https://github.com/vudangRVC/meta-renesas-sst/blob/styhead/rz-sbc-renesas-core-images/recipes-extend/systemd-network-control/files/quickboot-wayland/disable_networking_stack.sh#L10.

To unmask a service, we simply need to remove this null symlink.

We can also use commands to disable, mask, unmask, and restart services. However, a disadvantage that may affect the user experience is the execution time:
- Using systemctl to disable/mask and unmask/restart services for all services can take up to 20–25 seconds.
- Using manual symlinks takes only 1–2 seconds.

preetam-reddy replied:
Thats because you are not altering running systemd services. removing and adding links doesnt automatically trigger a service update. That still requires systemd calls which are not performed in the original implementation. The system working because of rebooting it.
Please use systemd commands.

tiennguyenuh replied:
Updated in the commit https://github.com/Renesas-SST/meta-renesas/commit/a5ec3ef984ea5fb8bf5d7b071e4528138421add3

preetam-reddy commented on recipes-extend/systemd-network-control/files/quickboot-wayland/enable_networking_stack.sh:
```diff
@@ -0,0 +1,204 @@
+    else
+        # Initialize to false if config file does not exist
+        WIFI_ACTIVE=false
+        BLUETOOTH_ACTIVE=false
```
Can you please run commands on the interface and supplicant to determine the actual status.

tiennguyenuh replied:
Updated in the commit: https://github.com/Renesas-SST/meta-renesas/commit/4fd7b7fe0947a1c577068d37a8f97a8170370a2f

tiennguyenuh committed:
rz-sbc: use systemctl command to detect services status

Use the systemctl command to detect services status instead of relying on
a file for tracking.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

tiennguyenuh committed:
rz-sbc: rename rzpi to rzg2l-sbc across all files and recipes

Rename all instances of `rzpi` to `rzg2l-sbc` within source files and recipes.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>

vudangRVC commented on recipes-extend/apt-sources-config/files/sources.list:
We're using Ubuntu 24.10 (Oracular Oriole) repository which might not compatible with the debian rootfs.
It's fine now but we need to address this when performing validation test later.
